src/main/java/org/dmfs/jems/fragile/composite/Zipped.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.fragile.composite;
+
+import org.dmfs.jems.fragile.Fragile;
+import org.dmfs.jems.function.BiFunction;
+
+
+/**
+ * A {@link Fragile} which combines two other {@link Fragile}s with a {@link BiFunction}.
+ *
+ * @author Marten Gajda
+ */
+public final class Zipped<Left, Right, Result, E extends Throwable> implements Fragile<Result, E>
```

Comment:
Can't you use generic type parameters for all 3 exceptions? Like (it compiles for me):
```java
public final class Zipped<Left, Right, Result, ResultException extends Throwable, LeftException extends ResultException, RightException extends ResultException> implements Fragile<Result, ResultException>
{
    private final BiFunction<Left, Right, Result> mZipFunction;
    private final Fragile<Left, LeftException> mLeft;
    private final Fragile<Right, RightException> mRight;


    public Zipped(Fragile<Left, LeftException> left, Fragile<Right, RightException> right, BiFunction<Left, Right, Result> zipFunction)
    {
        mZipFunction = zipFunction;
        mLeft = left;
        mRight = right;
    }


    @Override
    public Result value() throws ResultException
    {
        return mZipFunction.value(mLeft.value(), mRight.value());
    }
}
```

Reply:
Ok, but what's the advantage? We don't really care about `LeftException` and `RightException` other than that `ResultException` has to be a super type of both of them. They both can be anonymous, we don't need a name for them and the unit test seems to confirm that the concept works.

Reply:
Right, I overlooked that it's not just `?` but `? extends E` so the super type thing is enforced this way as well.
What was the slight issue then what you mentioned?

Reply:
`Intact` had a hard coded `Exception` as the generic `Throwable` parameter.

Reply:
Oh, I see, so that was a fix.